adaptive_breakpoints/lib/src/adaptive_breakpoints.dart

```diff
@@ -0,0 +1,300 @@
+      case AdaptiveWindowType.xl:
+        return RangeValues(1280, double.infinity);
+      default:
+        throw AssertionError('Error');
+    }
+  }
+
+  RangeValues get heightPortraitRangeValues {
+    switch (this) {
+      case AdaptiveWindowType.xs:
+        return RangeValues(0, 959);
+      case AdaptiveWindowType.s:
+        return RangeValues(360, 1599);
+      case AdaptiveWindowType.m:
+        return RangeValues(720, 1919);
+      case AdaptiveWindowType.l:
+        return RangeValues(1920, double.infinity);
+      case AdaptiveWindowType.xl:
+        return RangeValues(1920, double.infinity);
+      default:
+        throw AssertionError('Error');
+    }
+  }
+}
+
+class AdaptiveWindowLimits {
+  AdaptiveWindowLimits({
+    @required this.xs,
+    @required this.s,
+    @required this.m,
+    @required this.l,
+    @required this.xl,
+  })  : assert(xs != null),
+        assert(s != null),
+        assert(m != null),
+        assert(l != null),
+        assert(xl != null);
+
+  final int xs;
+  final int s;
+  final int m;
+  final int l;
+  final int xl;
+}
+
+/// This class represents the Material breakpoint system entry.
+/// https://material.io/design/layout/responsive-layout-grid.html#breakpoints
+class BreakpointSystemEntry {
+  const BreakpointSystemEntry({
+    @required this.range,
+    this.portrait,
+    this.landscape,
+    @required this.window,
+    @required this.columns,
+    @required this.margins,
+    @required this.gutters,
+  })  : assert(range != null),
+        assert(window != null),
+        assert(columns != null),
+        assert(margins != null),
+        assert(gutters != null);
+
+  /// The breakpoint range values represents a width range.
+  final RangeValues range;
+
+  /// String of device name that use this breakpoint range in portrait view.
+  final String portrait;
+
+  /// String of device name that use this breakpoint range in landscape view.
+  final String landscape;
+
+  /// Material generalizes the device size into five different windows: extra
+  /// small, small, medium, large, and extra large.
+  ///
+  /// The adaptive window represents a set of similar devices. For example, if
+  /// you want to create an adaptive layout for phones and small tablets you
+  /// would check if your window width is within the range of xs and s. If your
+  /// user has a bigger window size than you would create a different layout for
+  /// larger screens.
+  final AdaptiveWindowType window;
+
+  /// The number of columns in this breakpoint system entry.
+  /// https://material.io/design/layout/responsive-layout-grid.html#columns-gutters-and-margins
+  final int columns;
+
+  /// The margins in this breakpoint system entry. Margins is typically going to
+  /// be the same length as gutters.
+  /// https://material.io/design/layout/responsive-layout-grid.html#columns-gutters-and-margins
+  final double margins;
+
+  /// The size of gutters in pixels in this breakpoint system entry. Typically
```

Comment:
nit: reformat so that it's one sentence then newline then the rest

Reply:
Done